src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,134 @@ public function getTableRow(Element $element, $search)
+   */
+  public function assertTextInTableRows($text, $rowText) {
+    $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+    foreach ($rows as $row) {
+      if (str_contains($row->getText(), $text) !== false) {
+        throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+      }
     }
+  }
 
   /**
-   * Asset text not in a table row containing given text.
+   * Assert text not in a table row containing given text.
+   * If multiple rows are found, check the first one only.
    *
    * @Then I should not see (the text ):text in the :rowText row
    */
-    public function assertTextNotInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) !== false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+  public function assertTextNotInTableRow($text, $rowText)
+  {
+    $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
+    if (strpos($row->getText(), $text) !== false) {
+      throw new \Exception(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+    }
+  }
+
+  /**
+   * Asset text not in any table rows containing given text.
+   *
+   * @Then I should not see (the text ):text in any :rowText rows
+   */
+    public function assertTextNotInAnyTableRows($text, $rowText): void {
+        $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+        foreach ($rows as $row) {
+          if (strpos($row->getText(), $text) !== false) {
+            throw new \RuntimeException(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+          }
         }
     }
 
+  /**
+   * Assert text not in any table rows containing given text, or no table.
+   *
+   * @Then I should not see (the text ):text in any :rowText rows or no table
+   */
+  public function assertTextNotInTableRowOrNoTable($text, $rowText): void {
+    if ($this->hasTable($this->getSession()->getPage())) {
+      $this->assertTextNotInAnyTableRows($text, $rowText);
+    }
+  }
+
+  /**
+   * Find text in a table row containing given text, no text in a table row, or
+   * no table. This is used for the transient status.
+   *
+   * @Then I should see with (the text ):text in the :rowText row, no rows, or no table
+   */
+  public function assertNoTableRowOrSeeWithText($text, $rowText): void {
+    if ($this->hasTable($this->getSession()->getPage())) {
+      $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+      foreach ($rows as $row) {
+        if (strpos($row->getText(), $text) === false) {
+          throw new \RuntimeException(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+        }
+      }
+    }
+  }
```

Comment:
@kumikoono Let's use a guard clause:

```suggestion
  public function assertNoTableRowOrSeeWithText($text, $row_text): void {

    if (!$this->hasTable($this->getSession()->getPage())) {
      return;
    }

    $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
    foreach ($rows ?: [] as $row) {
      if (strpos($row->getText(), $text) === TRUE) {
        continue;
      }
      throw new \RuntimeException(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $row_text, $text));
    }
```